connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/GenericRecordTest.java

```diff
@@ -40,17 +44,73 @@
  */
 public class GenericRecordTest {
 
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testCorrectSerializerInitialization(Boolean useNestedRowsMode) {
+        GenericRecordToRowMutationSerializer serializer = createTestSerializer(useNestedRowsMode);
+
+        assertEquals(TestingUtils.ROW_KEY_FIELD, serializer.getRowKeyField());
+        assertEquals(
+                useNestedRowsMode ? null : TestingUtils.COLUMN_FAMILY,
+                serializer.getColumnFamily());
+        assertEquals(useNestedRowsMode, serializer.getUseNestedRowsMode());
+    }
+
     @Test
-    public void testCorrectRowMutationSerialization() {
-        GenericRecordToRowMutationSerializer serializer = createTestSerializer();
-        RowMutationEntry wantedEntry = TestingUtils.getTestRowMutationEntry();
+    public void testColumnFamilyAndNestedIncompability() {
+        Assertions.assertThatThrownBy(
+                        () ->
+                                GenericRecordToRowMutationSerializer.builder()
+                                        .withRowKeyField(TestingUtils.ROW_KEY_FIELD)
+                                        .withNestedRowsMode()
+                                        .withColumnFamily(TestingUtils.COLUMN_FAMILY)
+                                        .build())
+                .hasMessageContaining(ErrorMessages.COLUMN_FAMILY_AND_NESTED_INCOMPATIBLE);
+    }
 
-        GenericRecord record = getTestGenericRecord();
+    @Test
+    public void testNotColumnFamilyOrNestedRequired() {
+        Assertions.assertThatThrownBy(
+                        () ->
+                                GenericRecordToRowMutationSerializer.builder()
+                                        .withRowKeyField(TestingUtils.ROW_KEY_FIELD)
+                                        .build())
+                .hasMessageContaining(ErrorMessages.COLUMN_FAMILY_OR_NESTED_ROWS_REQUIRED);
```

Comment:
this test is making me think...should we just default to column family if nothing is passed?

Reply:
I don't think it's a good idea, since the column family would need to be passed. And we didn't want to have a default cf

Reply:
revive the "flink" default column family?

Reply:
Talked offline, we prefer users failing during jar packaging than having to check the error logs to find out they don't have CF `flink`